symbolic-base/src/ZkFold/Symbolic/Data/JWT/RS256.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module ZkFold.Symbolic.Data.JWT.RS256 (SigningKey (..), Certificate (..)) where
+
+import           Control.DeepSeq                    (NFData)
+import           GHC.Generics                       (Generic)
+import qualified Prelude                            as P
+
+import           ZkFold.Symbolic.Algorithms.RSA
+import           ZkFold.Symbolic.Class
+import           ZkFold.Symbolic.Data.ByteString    (ByteString)
+import           ZkFold.Symbolic.Data.Class
+import           ZkFold.Symbolic.Data.Input         (SymbolicInput)
+import           ZkFold.Symbolic.Data.JWT
+import           ZkFold.Symbolic.Data.VarByteString (VarByteString)
+
+-- | RSA Public key with Key ID
+--
+data Certificate ctx
+    = Certificate
+        { pubKid :: VarByteString 320 ctx
+        , pubKey :: PublicKey 2048 ctx
+        }
+    deriving Generic
+
+deriving instance
+    ( P.Eq (PublicKey 2048 ctx)
+    , P.Eq (VarByteString 320 ctx)
+    ) => P.Eq (Certificate ctx)
+deriving instance
+    ( P.Show (PublicKey 2048 ctx)
+    , P.Show (VarByteString 320 ctx)
+    ) => P.Show (Certificate ctx)
+deriving instance
+    ( NFData (PublicKey 2048 ctx)
+    , NFData (VarByteString 320 ctx)
+    ) => NFData (Certificate ctx)
+instance
+    ( SymbolicData (PublicKey 2048 ctx)
```

Comment:
Why is this constraint needed?

Reply:
To replace two constraints of the type `KnownNat (NumberOfRegisters ...)`. In every other instance where SymbolicData is a constraint the reasoning is the same